### PR TITLE
Changed database structure

### PR DESCRIPTION
**Pull request by JBK91**
- Removed `organization` column from `event` tables
- Created `app_user_role` table - one user may have role in more than 1 organisation
- Removed street address slitting from `organization` table - IMO we don't need 3 columns for street address (street, house and flat)
- created `animal_registration` table to store registration data and added `status` column to track if that registration is active
- created `animal_microchip` table to store microchip data and added `status` column to track if that chip is implanted
- created `animal_details` table to split animal general info from `animal` table

### CODE REVIEW

**Review by JustasBelevicius — APPROVED** [review]
Well done 👍 ! Since the product is not in production stages yet, it would be better to add all the schema changes to the 1-schema.sql editing the create queries directly. I know it wasn't done like that before, but I believe that would look cleaner until we are forced to do it all with alterations when the app is used.

tl;dr; 
- CREATE and ALTER (just edit table CREATION queries instead) can be moved to the 1-schema.sql
- INSERT queries go to 2-data.sql, it won't be run on production in the later stages of the app.

**Review by AzisK — APPROVED** [review]
Good job!

**Comment by AzisK** [comment]
> Well done 👍 ! Since the product is not in production stages yet, it would be better to add all the schema changes to the 1-schema.sql editing the create queries directly. I know it wasn't done like that before, but I believe that would look cleaner until we are forced to do it all with alterations when the app is used.

We can do this with a separate pull request since this approach is already used before

File: database/4-change_structure.sql [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+);
+
+INSERT INTO animal_details (animal_id, breed_id, name, gender, color, birth_date, weight, allergy, food) VALUES
+    (1, 205, 'Haskelis', 1, 61, '2020-08-01', 20, NULL, NULL),
+    (2, 268, 'Jupyteris', 1, 2, '2020-01-01', 1, NULL, NULL),
+    (5, 389, 'Murkė', 2, 4, '2019-09-16', 7, NULL, NULL),
+    (4, 350, 'Javainis', 1, 68, '2020-01-01', 0.4, NULL, NULL),
+    (3, 422, 'Pitoncas', 1, 32, '2020-01-01', 0.3, NULL, NULL);
+
+ALTER TABLE animal
```
Comment (psaulenas) [inline database/4-change_structure.sql:88]: Please update also https://github.com/pets-oss/pets-back/blob/main/src/schemas/animal.ts#L28

**Review by psaulenas — CHANGES_REQUESTED** [review]
You need also update GraphQL schema

File: database/4-change_structure.sql (hunk 1 above)
Comment (ignasposka) [reply database/4-change_structure.sql:88]: Why animal GQL schema should be updated? I think it should remain the same for the user of GQL endpoint 🤔

**Review by ignasposka — CHANGES_REQUESTED** [review]
I'm not sure what we are doing with animal type in GraphQL side. At this point I see removing lot of fields from animal type, but they still somehow should be available through GQL endpoint, what's the plan for that?

File: database/4-change_structure.sql (hunk 1 above)
Comment (JBK91) [reply database/4-change_structure.sql:88]: `created animal_registration table to store registration data and added status column to track if that registration is active
created animal_microchip table to store microchip data and added status column to track if that chip is implanted`

in the rare case microchips can be changed (damaged or something), registrations too. With these tables we will have historical data for animal. If that data will be only in animal where can be only one value, we lose historical data.

`animal_details` is to split general info/details from `animal` table

**Comment by JBK91** [comment]
> I'm not sure what we are doing with animal type in GraphQL side. At this point I see removing lot of fields from animal type, but they still somehow should be available through GQL endpoint, what's the plan for that?

I am not sure what needs for that, if frontend needs to have all data calling `animal`, than I will update animal query to include other related data, in other cases I will add endpoints.

IMO: `animal_details` could be in `animal` endpoint:
{
   animal {
         ....,
         details {
              ....
         }
   }
}

other (microchips and registrations) other endpoints

**Comment by ignasposka** [comment]
> > I'm not sure what we are doing with animal type in GraphQL side. At this point I see removing lot of fields from animal type, but they still somehow should be available through GQL endpoint, what's the plan for that?
> 
> I am not sure what needs for that, if frontend needs to have all data calling `animal`, than I will update animal query to include other related data, in other cases I will add endpoints.
> 
> IMO: `animal_details` could be in `animal` endpoint:
> {
> animal {
> ....,
> details {
> ....
> }
> }
> }
> 
> other (microchips and registrations) other endpoints

I think we could have GQL query `Animal` which would let to get animal with all microchips, details and registrations. This is the one of the main GQL feature, to let GQL client choose exactly what it wants. It would be silly to not use this advantage of GQL. In my opinion `Animal` query would look like the following:


```
{
    Animal {
        name,
        animal_details {
            color
        },
        animal_registration {
            status
        }
    }
}
```

**Review by ignasposka — APPROVED** [review]
Good to go, but please fix animal query after merging this.

File: src/sql-queries/animal.ts [hunk 2]
```diff
@@ -40,17 +27,6 @@ export const getAnimalsQuery = (): QueryConfig => {
                     registration_date,
```
Comment (psaulenas) [inline src/sql-queries/animal.ts:27]: You missed to remove `registration_no` and `registration_date`

**Review by psaulenas — APPROVED** [review]
Overall looks good, you missed to remove few columns from one query
